cmd/spiffe-aws-assume-role/cli/cli.go

```diff
@@ -62,22 +71,27 @@ func RunWithDefaultContext(args []string) error {
 	context := &CliContext{
 		JWTSourceProvider: credentials.StandardJWTSourceProvider,
 		STSProvider:       credentials.StandardSTSProvider,
+		Logger:            createLogger(),
 	}
 
 	return Run(context, args)
 }
 
-func Run(context *CliContext, args []string) error {
+func Run(context *CliContext, args []string) (err error) {
+	defer func() {
+		if err != nil {
+			context.Logger.Error(err)
+		}
+	}()
+
 	ctx, err := parse(args)
 	if err != nil {
-		return err
+		return errors.Wrap(err, fmt.Sprintf("failed to parse command line arguments: %s", args))
 	}
 
-	if err = ctx.Run(context); err != nil {
-		return err
-	}
+	err = ctx.Run(context)
 
-	return nil
+	return err
 }
```

Comment:
Really nice improvement over the previous revisions: combining named return values and defer to eliminate the need for a wrapper function to handle error logging.